pip/index.py

```diff
@@ -330,13 +335,16 @@ def mkurl_pypi_url(url):
                 ) or req.url_name
 
         if url_name is not None:
-            locations = [
-                mkurl_pypi_url(url)
-                for url in self.index_urls] + self.find_links
-        else:
-            locations = list(self.find_links)
+            return [mkurl_pypi_url(url) for url in self.index_urls]
+        return []
+
+    def _find_all_versions(self, req):
+        index_locations = self._get_index_urls_locations(req)
+        file_locations, url_locations = self._sort_locations(index_locations)
+        fl_file_loc, fl_url_loc = self._sort_locations(self.find_links)
```

Comment:
It's step for a potential future refactoring and also has the advantage to clearly state how each packages sources (index_urls, find_links and dependency_links) are dealt with.